src/mod/irc.mod/irc.c

```diff
@@ -776,6 +776,27 @@ static int invite_4char STDVAR
   return TCL_OK;
 }
 
+static int check_tcl_chghost(char *nick, char *from, char *mask, struct userrec *u,
+                             char *chan, char *ident, char * host)
+{
+  struct flag_record fr = { FR_GLOBAL | FR_CHAN | FR_ANYWH, 0, 0, 0, 0, 0 };
+  char usermask[UHOSTMAX];
+  int x;
+
+  get_user_flagrec(u, &fr, NULL);
+  simple_sprintf(usermask, "%s!%s@%s", nick, ident, host);
```

Comment:
also pls. use snprintf() instead. if compiler throws warning, resize the array. i know its often a mess with UHOSTMAX stuff, but can we have snprintf() everywhere from now on?